_i18n/en/resources/user-guides/mine-to-pool.md

```diff
@@ -1,107 +1,194 @@
+```
+cmake -Bbuild -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl
+make -Cbuild -j$(nproc)
+```
 
-This means that you can get around a 20% hashrate boost by enabling large pages.
+Copy binary and example configuration to your home directory:
 
-### Large pages on Linux
+```
+cp build/xmrig ~/
+cp src/config.json ~/
+```
 
-Firstly stop the miner (if it's running), run the following commands to enable
-large pages and then start the miner as root:
+## Selecting a pool
+
+There are lots of pools to choose from. You can find a list at
+[miningpoolstats.stream/monero](https://miningpoolstats.stream/monero).
+
+Choosing a larger pool means that you will see more frequent (but
+smaller) payouts, but choosing a smaller pools helps keep the network
+decentralised. [Miners will not lose any revenue by mining on a
+smaller pool](https://redd.it/g6uh2l).
+
+## Configuring the miner
+
+Navigate to your chosen pool's webpage and read their
+documentation. They should mention an address and port for you to
+enter into your miner. For example, `pool.xmr.pt:3333`.
+
+Next, open up the config.json you copied or extracted earlier in your
+favourite text editor. Scroll down to the line that says
+`donate.v2.xmrig.com:3333` and change the text inside the quotes to
+the address of your pool. The line below should contain
+`YOUR_WALLET_ADDRESS`. Change that to your actual wallet address.
+
+After these modifications, your configuration should look something like this:
+
+```
+{
+	// [...]
+	"pools": [
+		{
+			"url": "pool.xmr.pt:3333",
+			"user": "43YjW8SZov..."
+		}
+	],
+	// [...]
+}
+```
+
+## Starting the miner
+
+Windows users can double click on xmrig.exe. Users of other operating
+systems should `cd` into the directory that contains XMRig and then
+type `./xmrig` and press return.
+
+If you see green messages saying that shares have been accepted,
+congratulations, everything is working!
+
+## Troubleshooting
 
-	sudo sysctl -w vm.nr_hugepages=128
-	sudo ./xmr-stak-cpu
+### Anti-virus keeps removing XMRig
 
-### Large pages on Windows
+Some anti-viruses flag XMRig as malware because it is often deployed
+to infected computers to mine without the owner's consent. As it is
+your computer and you are configuring the miner to mine for you, it is
+safe to add XMRig to your anti-virus whitelist.
 
-Taken from `config.txt`:
+### Cannot read/set MSR
+
+On some CPUs, XMRig tries to increase performance by disabling certain
+features like your CPU's instruction prefetcher. These operations
+require root/administrator, so try right clicking xmrig.exe and
+running it as administrator, or running `sudo ./xmrig` on other
+systems.
+
+### Algo not known
+
+Find the line in config.json that says `algo: null` and change it to
+`algo: "rx/0"`. By default, XMRig expects the pool to tell it which
+hashing algorithm to use.
+
+### Huge Pages 0%
+
+#### Allowing large pages on Windows
+
+Taken from xmr-stak-cpu's documentation:
```

Comment:
Why was this added? Is the text below extracted from xmr-stak-cpu's documentation? If that's the case i would just rewrite it. It's known that fireice is hostile to Monero, i don't want to give him an excuse to complain or any visibility.

Reply:
It was in the original guide and I think I took it from xmr-stak-cpu's docs. Since it's just copied verbatim from [the MSDN](https://docs.microsoft.com/en-us/sql/database-engine/configure-windows/enable-the-lock-pages-in-memory-option-windows?view=sql-server-ver15) I'll get rid of the xmr-stak-cpu mention.